GitCommands/Git/GitModule.cs

```diff
@@ -3367,25 +3352,27 @@ internal GitBlame ParseGitBlame(string output, Encoding encoding)
                     GitBlameCommit commit;
                     if (hasCommitHeader)
                     {
-                        if (!commitByObjectId.ContainsKey(objectId))
+                        // TODO quite a few nullable suppressions here (via ! character) which should be addressed as they hint at a design flaw
+
+                        if (!commitByObjectId.ContainsKey(objectId!))
                         {
                             commit = new GitBlameCommit(
-                                objectId,
-                                author,
-                                authorMail,
+                                objectId!,
+                                author!,
+                                authorMail!,
                                 authorTime,
-                                authorTimeZone,
-                                committer,
-                                committerMail,
+                                authorTimeZone!,
+                                committer!,
+                                committerMail!,
                                 committerTime,
-                                committerTimeZone,
-                                summary,
-                                filename);
-                            commitByObjectId[objectId] = commit;
+                                committerTimeZone!,
+                                summary!,
+                                filename!);
```

Comment:
This doesn't look good.

Reply:
Agree. This was probably the worst example of logic that tripped up NRT. Honestly I think it should be re-written to be clearer and have fewer moving parts.